File: src/my_project/easy_problems/from151To200/shift_2d_grid.py
from typing import List 

class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        

        flat_list = list()
        
        x = len(grid)
        y = len(grid[0])
        solution = list()
        for i in range(x):
            for j in range(y):
                flat_list.append(grid[i][j])
        temp = None
        for i in range(k):
            temp = flat_list.pop()
            flat_list = [temp] + flat_list

        for i in range(x):
            solution.append(flat_list[i*y: i*y +y])
        return solution

solution = Solution()

print(solution.shiftGrid(grid = [[1,2,3],[4,5,6],[7,8,9]], k = 1))

print(solution.shiftGrid( grid = [[3,8,1,9],[19,7,2,5],[4,6,11,10],[12,0,21,13]], k = 4))

print(solution.shiftGrid( grid = [[13]], k = 4))

File: my_project_typescript/easy_problems_typescript/from51To100/shift2DGrid.ts
function shiftGrid(grid: number[][], k: number): number[][] {
    let flatList: Array<number> = []

    let x: number = grid.length
    let y: number = grid[0].length
    let solution: Array<any> = []

    for (let i = 0; i < x; i++){
        for (let j = 0; j < y; j++){
            flatList.push(grid[i][j])
        }
    }
    let temp: any = 0
    for (let i = 0; i < k; i++){
        temp = flatList.pop()
        flatList = [temp].concat(flatList)
    }

    for (let i = 0; i<x; i++){
        solution.push(flatList.slice(i*y, i*y+y))
    }
    return solution
};


console.log(shiftGrid([[3,8,1,9],[19,7,2,5],[4,6,11,10],[12,0,21,13]], 4))